iothub/device/src/Pipeline/ConnectionStateDelegatingHandler.cs

```diff
@@ -99,7 +101,7 @@ public override async Task OpenAsync(CancellationToken cancellationToken)
                                             Logging.Error(this, ex, nameof(HandleDisconnectAsync));
 
                                         HandleConnectionStatusExceptions(ex, true);
-                                        if (_clientTransportStateMachine.GetCurrentState() == ClientTransportState.Opening)
+                                        if (_clientTransportStateMachine.GetCurrentState() == ClientTransportState.Opening && !_wasOpened)
```

Comment:
Cheaper operation first.

```suggestion
                                        if (!_wasOpened && _clientTransportStateMachine.GetCurrentState() == ClientTransportState.Opening)
```